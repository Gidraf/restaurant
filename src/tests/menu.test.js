import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure, mount } from 'enzyme';
import MenuComponent from '../components/Menu';

configure({ adapter: new Adapter() });


describe('Menu Component', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <MenuComponent />,
    );
  });

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('contains main intro div', () => {
    const div = wrapper.find('div');
    expect(div).toHaveLength(3);
  });
  it('contains main buttons', () => {
    const button = wrapper.find('button');
    expect(button).toHaveLength(1);
  });

  it('Render menu items', () => {
    const button = wrapper.find('button');
    wrapper.setState({
      imageClassname: 'show-menu',
    });
    wrapper.instance().animateMenu();
    expect(button).toHaveLength(1);
  });
  it('contains main buttons', () => {
    const div = wrapper.find('.fadetext');
    expect(div).toHaveLength(1);
  });
});
