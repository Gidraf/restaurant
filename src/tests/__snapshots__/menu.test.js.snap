// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockModal Component renders properly 1`] = `undefined`;

exports[`Menu Component renders properly 1`] = `ShallowWrapper {}`;
